src/query/util/opentracing/context.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package opentracingutil provides more utilities for dealing with opentracing and context.Context's.
+package opentracingutil
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+)
+
+// alias GlobalTracer() so we can mock out the tracer without impacting tests outside the package
+var getGlobalTracer = opentracing.GlobalTracer
+
+// SpanFromContextOrRoot is the same as opentracing.SpanFromContext, but instead of returning nil, it starts a
+// new root span if ctx doesn't already have an associated span, instead of returning nil. Use this over
+// opentracing.StartSpanFromContext if you need access to the current span, but don't want to start a child span.
+// NB: the span returned by this function won't really be configured, and won't be attached to the context; if you
+// want a proper span, start one at the root and pass it in.
+func SpanFromContextOrRoot(ctx context.Context) opentracing.Span {
+	sp := opentracing.SpanFromContext(ctx)
+	if sp != nil {
+		return sp
+	}
+
+	return opentracing.StartSpan("SpanFromContextOrRoot - dummy")
```

Comment:
Where does this string appear re: trace? Could be good to make it a constant if nothing else